### ISSUE TITLE
Determining if consumer_key/secret exist

### ISSUE THREAD

**Issue by michaelhelmick** [maraujop/requests-oauth#18]
Say I set

``` python
OAuthHook.consumer_key = 'my_consumer_key'
OAuthHook.consumer_secret = 'my_consumer_secret'
```

It'd be nice if there was a way to determine if consumer_key and consumer_secret were set.

In my case, I want to use a request.session with a hook-pre_request of OAuthHook() using the key/secret if they are set AND if the oauth_token/secret is set when someone initializes my class, I want to then have request.session use the hook-pre_request also passing the oauth_token/secret

Let me know if you need me to elaborate more?

Something like:

``` python
if OAuthHook.consumer_key is not None and OAuthHook.consumer_secret is not None:
    self.client = requests.session(hooks={'pre_request': OAuthHook()})

if oauth_token and oauth_secret:  # These are the ones that are passed in my init from maybe a recored in the db
    oauth_hook = OAuthHook(self.oauth_token, self.oauth_secret)
    self.client = requests.session(hooks={'pre_request': oauth_hook})
```

**Comment by maraujop** [comment]
Sorry Mike,

I've read this several times and I'm not sure what the question is or what do you want me to do. Remember that the hook doesn't do any connection handling, your binding should take care of this. The hook is coded for let you sign your requests with OAuth1.0.  It only intercepts your requests and "signs" them.

Anything beyond this, is something that lives in a separate wrapper on top. Does this seem reasonable? 

BTW did you get to test `dev` branch. I'm quite sure it should fix your issue.

Cheers,
Miguel

**Comment by michaelhelmick** [comment]
No. I've been busy and haven't got to test. 

As far as the issue. I just wanted a way to determine of OAuthHook.consumer_key and OAuthHook.consumer_secret were set.

**Comment by michaelhelmick** [comment]
If OAuthHook.consumer_key isn't set. Currently, it won't return None

**Comment by maraujop** [comment]
Hi Mike,

I've set `None` for default values for `consumer_key` and `consumer_secret`. Sorry, to take so long, I should have understood this the first time, your explanation was more than good.

Cheers,
Miguel

**Comment by michaelhelmick** [comment]
Awesome, thanks!

**Comment by maraujop** [comment]
No problem, thank you!

**Comment by maraujop** [comment]
Version 0.4.0 has been released, it works well with requests 0.12.1 or higher. I'm closing this issue.

Cheers
